### PR TITLE
Optionally rotate wind/stress from to computational grid (from true north)

### PR DESCRIPTION
**Pull request by mhrib**
For detailed information about submitting Pull Requests (PRs) to the CICE-Consortium,
please refer to: <https://github.com/CICE-Consortium/About-Us/wiki/Resource-Index#information-for-developers>

## PR checklist
- [x ] Short (1 sentence) summary of your PR: 
Optional rotate wind vectors to computational grid (default)
- [ ] Developer(s): 
    Mads Hvid Ribergaard, DMI
- [ ] Suggest PR reviewers from list in the column to the right.
- [ ] Please copy the PR test results link or provide a summary of testing completed below.
    Able to merge. These branches can be automatically merged.
- How much do the PR code changes differ from the unmodified code? 
    - [x ] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [x ] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [ x] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [x ] Yes
    - [ ] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [ ] No 
- [x ] Please provide any additional information or relevant details below:

A completely other change within this pull request:
* Also printout seconds, if a reading error occur reading "hycom"-forcing within ice_forcing.F90

### CODE REVIEW

File: doc/source/cice_index.rst [hunk 1]
```diff
@@ -91,6 +91,7 @@ either Celsius or Kelvin units).
    "bTiz", "temperature of ice layers on bio grid", ""
    "**C**", "", ""
    "c<n>", "real(\ :math:`n`)", ""
+   "rotate_wind", ":math:`\bullet` if true, rotate wind/stress composants to computational grid", "T"
```
Comment (eclare108213) [inline doc/source/cice_index.rst:94]: please fix typo 'components'

File: cicecore/cicedynB/general/ice_forcing.F90 [hunk 2]
```diff
@@ -1626,12 +1629,20 @@ subroutine prepare_forcing (nx_block, ny_block, &
 
       endif
 
-      if (calc_strair) then
+      if (calc_strair .and. (.not.rotate_wind)) then
```
Comment (eclare108213) [inline cicecore/cicedynB/general/ice_forcing.F90:1632]: The logic here is hard to follow.  Maybe better like this?  
```
if (calc_strair) then
  if (rotate_wind) then
     ...
  else ! calc_strair and wind not rotated
     ...
  endif
else ! .not. calc_strair
  if (rotate_wind) then
     ...
  else ! not calc_strair and wind not rotated - not currently implemented
     ...
  endif
endif
```

File: doc/source/cice_index.rst (hunk 1 above)
Comment (mhrib) [reply doc/source/cice_index.rst]: Corrected

File: cicecore/cicedynB/general/ice_forcing.F90 (hunk 2 above)
Comment (mhrib) [reply cicecore/cicedynB/general/ice_forcing.F90]: Agree, It is easier to read splitting up like you suggested despite of only 3 possible outcome ... today. I have repeated the do-loops to avoid if-sentence inside loops.

**Comment by apcraig** [comment]
How much testing has been done?  Are we confident it's all bit-for-bit?

**Comment by mhrib** [comment]
@apcraig w.r.t. bit-for-bit. Depending on rotate_wind option (true/false) it will skip the rotation of wind/stress. So it assumes, that the provided forcing is rotated correctly  and exactly like the default inline rotation.

**Comment by apcraig** [comment]
@mhrib I understand what the code is supposed to do.  I just want to confirm that you have compiled and run this code, and that the default setting produces bit-for-bit results.  That information was not included in the PR information yet.  Thanks.

**Comment by mhrib** [comment]
> @mhrib I understand what the code is supposed to do. I just want to confirm that you have compiled and run this code, and that the default setting produces bit-for-bit results. That information was not included in the PR information yet. Thanks.

@apcraig Sorry, I did not perform the test (which I always should do!). Now tested with intel compiler on GX3 grid (4x4) running for one day (24 steps) using the same ice_in file without rotate_wind defined (ie. set to default value = .true.): md5sum's on restart-files are equal.
fde1877b90dcc38e160d3ee1a520217c  restart/iced.1998-01-02-00000.nc_new
fde1877b90dcc38e160d3ee1a520217c  restart/iced.1998-01-02-00000.nc_org

**Comment by apcraig** [comment]
Thanks @mhrib, I also ran a full test suite last night and everything looks good.  https://github.com/CICE-Consortium/Test-Results/wiki/cice_by_hash#3fedc78b1b012f8bd6c0fde7c1b41e79b9db63a9
